contracts/keep3r/VaultKeep3r.sol

```diff
@@ -0,0 +1,135 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.8;
+
+import '@openzeppelin/contracts/math/SafeMath.sol';
+
+import '../../interfaces/keep3r/IVaultKeep3r.sol';
+import '../../interfaces/yearn/IEarnableVault.sol';
+
+import '../utils/Governable.sol';
+import '../utils/CollectableDust.sol';
+
+import './Keep3rAbstract.sol';
+
+contract VaultKeep3r is Governable, CollectableDust, Keep3r, IVaultKeep3r {
+  using SafeMath for uint256;
+  
+  mapping(address => uint256) public requiredEarn;
+  mapping(address => uint256) public lastEarnAt;
+  uint256 earnCooldown;
+
+  EnumerableSet.AddressSet internal availableVaults;
+
+  constructor(
+    address _keep3r,
+    address _bond,
+    uint256 _minBond,
+    uint256 _earned,
+    uint256 _age,
+    uint256 _earnCooldown
```

Comment:
Why do you need cooldown exactly?

Reply:
Explained on Telegram.
TLDR: for reducing the risk of getting the job credit drained.